svm/src/account_loader.rs

```diff
@@ -95,6 +95,262 @@ pub struct FeesOnlyTransaction {
     pub fee_details: FeeDetails,
 }
 
+#[derive(Debug, Clone)]
+pub(crate) struct AccountLoader<'a, CB: TransactionProcessingCallback> {
+    pub program_cache: ProgramCacheForTxBatch,
+    account_overrides: Option<&'a AccountOverrides>,
+    account_cache: HashMap<Pubkey, AccountCacheItem>,
+    callbacks: &'a CB,
+}
```

Comment:
as-written this struct needs to be passed around mutably. `account_overrides` and `callbacks` are never mutated, but `account_cache` must be mutated by `AccountLoader` functions, and `program_cache` by external functions. if youd rather we use an interior mutability pattern for these, please let me know what you suggest